### PR TITLE
NMS-14349: Add terms to glossary.adoc

### PR DESCRIPTION
**Pull request by emarsh1995**
Added entries for Traps, Daemon, Availability, Observability, Topology, Telemetry, Auto-discovered nodes, Direct-discovered nodes, and Passive discovery.

### All Contributors

* [x] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://issues.opennms.org) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the JIRA issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here, or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready. Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): https://issues.opennms.org/browse/NMS-14349

### CODE REVIEW

File: docs/modules/reference/pages/glossary.adoc [hunk 1]
```diff
@@ -127,9 +138,15 @@ Notifications:: Messages delivered to a set of recipients.
 {page-component-title} notifications inform users about events in their monitored networks without forcing them to log in and look at the UI.
 (See xref:operation:notifications/introduction.adoc[Notifications].)
 
+Observability:: The ability to measure the internal state of a system by examining its output.
+If the system's state can be estimated using only its output, it is considered "observable".
+
 Operator board:: Customizable dashboard to visualize monitoring information.
 (See xref:operation:admin/webui/opsboard/introduction.adoc[Operator Board].)
 
+Passive discovery:: The process by which a ``newSuspect`` event is detected by the {page-component-title} Provisiond service and converted into a new node.
```
[the remark this answers is not in the document]
Comment (mmahacek) [reply docs/modules/reference/pages/glossary.adoc:147]: Personal note here, but it's strongly recommended NOT to use this, as it just adds the nodes to the database and doesn't give an option to get the nodes into a requisition, which makes it harder to manage them in the long run.

File: docs/modules/reference/pages/glossary.adoc [hunk 2]
```diff
@@ -158,6 +175,14 @@ It also supports thresholding for streaming telemetry if you are using the xref:
...
 Time series, time-series database (TSDB):: Time series is a sequence of data points that occur in successive order over a period of time.
 A time series database (TSDB) is designed to store and serve time series data.
-(See xref:deployment:time-series-storage/timeseries/time-series-storage.adoc[Time Series Storage] and https://en.wikipedia.org/wiki/Time_series_database[Time Series Database].)
\ No newline at end of file
+(See xref:deployment:time-series-storage/timeseries/time-series-storage.adoc[Time Series Storage] and https://en.wikipedia.org/wiki/Time_series_database[Time Series Database].)
+
+Topology:: A description of a network's elements (devices, services) and the connections between them.
+(See xref:operation:topology/introduction.adoc[Introduction to Topology].)
+
+Traps:: Event triggers collected from SNMP-capable devices on the network by the {page-component-title} Trapd service daemon.
```
Comment (mmahacek) [inline docs/modules/reference/pages/glossary.adoc:187]:
```suggestion
Traps:: Event triggers generated by SNMP-capable devices on the network and send to the {page-component-title} Trapd service daemon.
```

File: docs/modules/reference/pages/glossary.adoc [hunk 3]
```diff
@@ -158,6 +175,14 @@ It also supports thresholding for streaming telemetry if you are using the xref:
 Snaps:: Self-contained software packages that run in a sandbox and have mediated access to host systems.
 Our Appliance Service uses snaps to distribute operating system packages and Minion service updates.
 
+Telemetry:: The process of automatically recording and transmitting data from an external source to a system in a different location for monitoring or analysis.
```
Comment (mmahacek) [inline docs/modules/reference/pages/glossary.adoc:178]:
```suggestion
Telemetry:: The process of automatically recording and transmitting data from an external source to a system in a different location for monitoring or analysis.
The Telemetry daemon in {page-component-title} accepts data sent from nodes in one of the supported protocol formats.
(See xref:operation:telemetryd/introduction.adoc[Telemetry Daemon].)
```

File: docs/modules/reference/pages/glossary.adoc (hunk 2 above)
Comment (emarsh1995) [reply docs/modules/reference/pages/glossary.adoc:187]: Just fixing a minor tense error in the suggestion. :) 

```suggestion
Traps:: Event triggers generated by SNMP-capable devices on the network and sent to the {page-component-title} Trapd service daemon.
```

File: docs/modules/reference/pages/glossary.adoc [hunk 4]
```diff
@@ -127,9 +139,16 @@ Notifications:: Messages delivered to a set of recipients.
 {page-component-title} notifications inform users about events in their monitored networks without forcing them to log in and look at the UI.
 (See xref:operation:notifications/introduction.adoc[Notifications].)
 
+Observability:: The ability to measure the internal state of a system by examining its output.
+If the system's state can be estimated using only its output, it is considered "observable".
```
Comment (Bonrob2) [inline docs/modules/reference/pages/glossary.adoc:143]: This is a nice addition!

Comment (Bonrob2) [inline docs/modules/reference/pages/glossary.adoc:143]:
```suggestion
If the system's state can be estimated using only its output, it is considered "observable".
```
This is a nice addition!

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
Just one change. Thanks!

**Review by mmahacek — APPROVED** [review]

**Pull request merged**